packages/nextjs-mf/src/plugins/container/InvertedContainerPlugin.ts

```diff
@@ -86,6 +86,113 @@ class InvertedContainerPlugin {
           'InvertedContainerPlugin',
           handler
         );
+        compilation.hooks.afterOptimizeChunks.tap(
+          'InvertedContainerPlugin',
+          (chunks) => {}
+        );
+        compilation.hooks.afterOptimizeChunkModules.tap(
+          'InvertedContainerPlugin',
+          (chunks, modules) => {
+            for (const chunk of chunks) {
+              if (
+                chunk.hasRuntime() &&
+                chunk.name === this.options?.container
+              ) {
+                //remove module from chunk
+                //@ts-ignore
+                // modules.forEach(chunk.removeModule);
+                const getmodules = chunk.getModules();
+
+                console.log(getmodules.length, chunk.name);
+              }
+            }
+          }
+        );
+        compilation.hooks.optimizeChunkModules.tap(
+          'InvertedContainerPlugin',
+          (chunks, modules) => {
+            for (const chunk of chunks) {
+              if (
+                chunk.hasRuntime() &&
+                chunk.name === this.options?.container
+              ) {
+                const eagerModulesInRemote = new Set();
+
+                for (const module of modules) {
+                  if (!compilation.chunkGraph.isModuleInChunk(module, chunk)) {
+                    continue;
+                  }
+
+                  if (
+                    module.type === 'provide-module' ||
+                    module.type === 'consume-shared-module'
+                    // || module.constructor.name === 'NormalModule'
+                  ) {
+                    // if its a shared module and is eager, we want to remove that eager reference
+                    //@ts-ignore
+                    if (module._eager) {
+                      //@ts-ignore
+                      eagerModulesInRemote.add(module._request);
+                    }
+                    //     console.log(module);
+                    //@ts-ignore
+                    if (module?._eager || module?._name?.startsWith('next')) {
```

Comment:
if its eager remove the connection from the remoteEntry, also if it starts with next, remove the registration connection code. Remotes will now not offer any next internals to share scope.